src/audio/module_adapter/module_adapter.c

```diff
@@ -1450,9 +1450,9 @@ int module_adapter_reset(struct comp_dev *dev)
 #endif /* CONFIG_ZEPHYR_DP_SCHEDULER */
 	if (IS_PROCESSING_MODE_SINK_SOURCE(mod)) {
 		/* for both LL and DP processing */
-		for (int i = 0; i < mod->num_of_sources; i++)
+		for (i = 0; i < mod->num_of_sources; i++)
```

Comment:
@singalsu @iuliana-prodan did we not recently change this in another place ? I think the newer xtensa compilers are possibly stricter here and hence we are hitting this now.